src/udev/net/ethtool-util.c

```diff
@@ -206,3 +212,116 @@ int ethtool_set_wol(int *fd, const char *ifname, WakeOnLan wol) {
 
         return 0;
 }
+
+static int ethtool_get_stringset(int *fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **gstrings) {
+        _cleanup_free_ struct ethtool_sset_info *info = NULL;
+        _cleanup_free_ struct ethtool_gstrings *strings = NULL;
+        unsigned len;
+        int r;
+
+        info = malloc0(sizeof (*info) + sizeof (unsigned));
```

Comment:
Hmm, please write `sizeof(struct ethtool_sset_info)` here. Dereferencing `*info` here doesn't look right, given that you assign NULL to it.